src/lib.rs

```diff
@@ -274,10 +275,19 @@ impl<T: Runtime> Client<T> {
         &self,
         key: StorageKey,
         hash: Option<T::Hash>,
+        modifier: StorageEntryModifier,
     ) -> Result<Option<V>, Error> {
-        if let Some(data) = self.rpc.storage(&key, hash).await? {
-            Ok(Some(Decode::decode(&mut &data.0[..])?))
+        if let Some(mut data) = self.rpc.storage(&key, hash).await? {
+            let bytes = if modifier == StorageEntryModifier::Optional {
+                let mut bytes = vec![1u8];
```

Comment:
If I'm reading this correctly then this will always return `Some` for `StorageEntryModifier::Optional` values - is that correct?